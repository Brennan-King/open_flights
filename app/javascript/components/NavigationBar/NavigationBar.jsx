import React from 'react';
import {Link} from 'react-router-dom';
import airplane from '../../images/airplane.png';
import './NavigationBar.scss';

const openFlightsText = 'Open Flights';
const homeText = 'Home';
const aboutUsText = 'About Us';

/**
 * Constructs a navigation bar component used to navigate pages.
 * @return {node} a div containing the configured navigation bar
 */
const NavigationBar = () => {
  return (
    <div className="navigation-bar-container">
      <div className="navigation-bar-open-flights-text">{openFlightsText}</div>
      <img
        className="navigation-bar-airplane-image"
        src={airplane}
        alt="airplane"
      />
      <div className="link-container">
        <Link className="home-link" to="/">{homeText}</Link>
        <Link className="about-us-link" to="/about-us">{aboutUsText}</Link>
      </div>
    </div>
  );
};

export default NavigationBar;
