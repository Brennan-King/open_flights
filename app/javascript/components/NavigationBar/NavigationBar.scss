.navigation-bar-container {
  display: flex;
  height: 50px;
  width: 100%;
  background: #b1b1b1;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: 30px;
  font-weight: bold;
  color: #fff;
  border: 3px solid black;
  z-index: 100;
}

.navigation-bar-open-flights-text {
  padding-left: 25px;
  padding-top: 8px;
}

.navigation-bar-airplane-image {
  padding-left: 5px;
}

.link-container {
  padding-left: 25px;
  padding-top: 8px;
  padding-right: 5px;
  margin-left: auto;
}

.home-link {
  text-decoration: none;
  color: #fff;
  padding-right: 20px;
}

.about-us-link {
  text-decoration: none;
  color: #fff;
  padding-right: 10px;
}