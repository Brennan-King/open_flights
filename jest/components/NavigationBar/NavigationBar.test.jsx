/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import NavigationBar from '../../../app/javascript/components/NavigationBar/NavigationBar';
import {BrowserRouter, Router} from 'react-router-dom';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for NavigationBar component
 */
describe('NavigationBar', () => {
  it('should render a NavigationBar component', () => {
    const component = mount(
        <Router history={new BrowserRouter().history}>
          <NavigationBar />
        </Router>,
    );

    expect(component).toMatchSnapshot();
  });
});
