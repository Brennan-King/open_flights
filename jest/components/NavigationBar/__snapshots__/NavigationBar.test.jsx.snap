// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should render a NavigationBar component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NavigationBar>
    <div
      className="navigation-bar-container"
    >
      <div
        className="navigation-bar-open-flights-text"
      >
        Open Flights
      </div>
      <img
        alt="airplane"
        className="navigation-bar-airplane-image"
        src="test-file-stub"
      />
      <div
        className="link-container"
      >
        <Link
          className="home-link"
          to="/"
        >
          <LinkAnchor
            className="home-link"
            href="/"
            navigate={[Function]}
          >
            <a
              className="home-link"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="about-us-link"
          to="/about-us"
        >
          <LinkAnchor
            className="about-us-link"
            href="/about-us"
            navigate={[Function]}
          >
            <a
              className="about-us-link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </div>
  </NavigationBar>
</Router>
`;
